app/services/update_account_service.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class UpdateAccountService < BaseService
+  def call(account, params)
+    before_locked = account.locked
```

Comment:
Minor nitpick: would call this `was_locked`

Then the conditional reads "if was locked and is not locked" which sounds better